### ISSUE TITLE
Log refactoring

### ISSUE THREAD

**Issue by leroyloren** [lavinir/hassio-onedrive-backup#51]
Would it be possible to modify the log writing logic?

Model example:
last couple of days I solved that the backup is not being created, which should take place at 2am, when I want to find out in the morning where the problem is I only see 1h old information in the log like every 5 min.

2/19/2023 09:11:59 [34] Info: Checking backups
02/19/2023 09:11:59 [34] Info: Retrieving existing local backups...
02/19/2023 09:11:59 [34] Info: Retrieving existing online backups...
02/19/2023 09:11:59 [34] Info: Last local backup Date: 02/19/2023 02:04:38
02/19/2023 09:11:59 [34] Info: Last online backup Date: 02/19/2023 02:04:38
02/19/2023 09:11:59 [34] Info: Online backups synced. No upload required

This makes it impossible to really know what happened, so I suggest not logging (optionally) internal checks and only log backup related information.

**Comment by lavinir** [comment]
I wouldn't want to drop those logs as they are an indication things are executing properly. I will add an option to change desired log verbosity though optionally so you only log warning / error if you wish.

**Comment by leroyloren** [comment]
Great, that sounds good.

The current state is like a debug with an hourly history :-)

Personally I like the log outputs like:

![obrazek](https://user-images.githubusercontent.com/57643470/221362434-ae5a70aa-ab58-4783-bb0f-281533bb683d.png)

**Comment by lavinir** [comment]
Configuring log verbosity is now available in version 2.0

**Comment by leroyloren** [comment]
I've tried, but now the behavior is that I don't know anything :-) I switched to warning, in general I would appreciate to see in the log in case of info when the addon is doing realistically what I want it to do, i.e. backup. No need to see internal checks every 5 min.

My idea is to see what the addon does at my allowed times, the current version of the info now behaves more like a debug.

**Comment by lavinir** [comment]
Yes you have a point, I'll make a pass over the logs see where adjustments make sense.

**Comment by lavinir** [comment]
fixed in 2.02

**Comment by leroyloren** [comment]
https://github.com/lavinir/hassio-onedrive-backup/blob/f50247b2caa89ebd447e2595eeefa4cf00f11630/onedrive-backup/Graph/GraphHelper.cs#L159

I think it might be Verbose too. I really think that in se I need to know from the log the creation and successful upload to the remote. Certainly not the progress of the uploading.

**Comment by lavinir** [comment]
Agreed.. Didn't make a pass through all the logs so I imagine some "incorrect" ones will pop up.. 
I will update the progress logs as well.

**Comment by lavinir** [comment]
Updated in 2.1
